src/palm-react-native-ui-kit/screens/channels/ChannelInfoScreen.tsx

```diff
@@ -45,8 +47,26 @@ const ChannelInfoScreen = (): ReactElement => {
   const { channel, channelName, tags, desc, gatingToken, loading } =
     useChannelInfo({ channelUrl: params.channelUrl })
 
-  const channelMembers =
-    channel?.members.sort(a => (a.profileUrl ? -1 : 1)) || []
+  const [channelMembers, setChannelMembers] = useState<Member[]>([])
+
+  const isFocused = useIsFocused()
+
+  useEffect(() => {
+    if (isFocused) {
+      channel?.refresh()
+    }
+  }, [isFocused])
+
+  useEffect(() => {
+    if (channel?.members) {
+      const sortedMembers = channel.members.sort(a => (a.profileUrl ? -1 : 1))
+      const moveToFirstMembers = UTIL.moveToFirstOfArray(
+        sortedMembers,
+        member => (member.metaData as SbUserMetadata).address === user?.address
+      )
+      setChannelMembers(moveToFirstMembers)
+    }
+  }, [channel?.members])
```

Comment:
is this really the best way to do this? we are interating through the members list every single time when a new user joins.

I would rather not unless necessary.

Reply:
Previously, members were sorted every time they were rendered.
this changed to sort only when there is a change in member.
And in some chat apps, there is a case where show ourself at the top, so this is added.

I will delete this method if it is not needed.

Reply:
deleted.